docs/_tooling/extensions/score_metamodel/metamodel.yaml

```
@@ -396,3 +396,29 @@ needs_extra_links:
   implements:
     incoming: "implemented by"
     outgoing: "implements"
+
+##############################################################
+# Graph Checks
+# The graph checks focus on the relation of the needs and their attributes.
+# Checks are defined in the following way:
+# - apply: defines the needs types to which the check should be applied
+#   - [include / exclude]: need types to which the check should be applied;
+#                          to perform the check on all needs types, set include to "."
+#   - condition: defines (together with apply) the condition which the needs need to fulfill
+#     - [and / or / xor / not]
+# - check: defines the check that should be applied
+#   - relation: defines the relation which is used derive the type of the parent need
+#   - condition: defines the condition that should be checked
+#     - [and / or / xor / not]
+##############################################################
+graph_checks:
+  safety_check:
+    apply:
```

```suggestion
    when:
```
Is this the when clause?

---

It's the same with the other parameters I'm afraid. They are not as self explanatory as you hoped ;-)

---

Maybe we can change the yaml to:

```
needs:
  include: comp_req
  condition: safety != QM
check:
  and:
    satisfies: safety != QM
    satisfies:  status == valid 
```

---

This does not state that there must an satisfies link at all.

---

This is checked by the mandatory attributes check

---

Mhh, interesting. So we have split the checks for the same topic across different technological aspects. That's a discussion for the future.